pennylane/resource/error.py

```diff
@@ -77,3 +78,68 @@ def error(self) -> AlgorithmicError:
         Returns:
             AlgorithmicError: The error.
         """
+
+
+class SpectralNormError(AlgorithmicError):
+    """Class representing the spectral norm error.
+
+    The spectral norm error is defined as the distance (in spectral norm) between the true unitary
+    we wish to apply and the approximate unitary that we actually apply.
+
+    Args:
+        error (float): The numerical value of the error
+
+    **Example**
+
+    >>> s1 = SpectralNormError(0.01)
+    >>> s2 = SpectralNormError(0.02)
+    >>> s3 = s1.combine(s2)
+    >>> print(type(s3), s3.error)
+    <class 'pennylane.resource.error.SpectralNormError'> 0.03
```

Comment:
What is the repr of `SpectralNormError`? What do we want it to be? I kind of expect
```
>>> s1.combine(s2)
<some nice repr>
```

Reply:
Added this:

```
>>> s1.combine(s2)
<SpectralNormError(0.03)>
```

Reply:
Much better @soranjh, thank you for that.